WordPress/gutenberg
# Align rename modals

Follow-up to #62788.

The vast majority of rename modals appear in the following format:

<img width="423" alt="Screenshot 2024-06-26 at 12 08 04" src="https://github.com/WordPress/gutenberg/assets/846565/09788512-0d4d-4cb1-845b-4cec1dd4b614">

* Succinct title
* Visible "Name" label
* "Save" button
* 40px buttons

This PR brings a couple of outliers into alignment with this convention.

### Navigation menus

| Before | After |
| --- | --- |
| <img width="514" alt="Screenshot 2024-06-26 at 12 10 18" src="https://github.com/WordPress/gutenberg/assets/846565/de925f5a-ff4f-4186-941f-0dea2b9531e8"> | <img width="430" alt="Screenshot 2024-06-26 at 14 04 27" src="https://github.com/WordPress/gutenberg/assets/846565/71b40c89-3d85-4369-9670-a9d33125e001"> |

Note the visible label. 

### Renaming blocks

| Before | After |
| --- | --- |
| <img width="477" alt="Screenshot 2024-06-26 at 12 09 19" src="https://github.com/WordPress/gutenberg/assets/846565/ff385188-8553-47bb-a507-ea8a91401b9b"> | <img width="494" alt="Screenshot 2024-06-26 at 14 06 10" src="https://github.com/WordPress/gutenberg/assets/846565/c6ac223e-68b2-447c-923b-be5ace2258fc"> |
| <img width="467" alt="Screenshot 2024-06-26 at 12 09 30" src="https://github.com/WordPress/gutenberg/assets/846565/bb7c6325-e7d0-47b9-9c74-57c65b1dd2be"> | <img width="437" alt="Screenshot 2024-06-27 at 11 11 58" src="https://github.com/WordPress/gutenberg/assets/846565/8d0eb1d9-22f2-4de5-b0f4-342ed1566eae"> |


### Renaming custom views
| Before | After |
| --- | --- |
| <img width="414" alt="Screenshot 2024-06-26 at 12 58 28" src="https://github.com/WordPress/gutenberg/assets/846565/d4c2d14e-47f8-448f-9f1c-e111f7313f54"> |  <img width="436" alt="Screenshot 2024-06-26 at 14 14 39" src="https://github.com/WordPress/gutenberg/assets/846565/dca9157e-a0cd-421d-9770-cd4b7043e094"> |

* Changed modal heading
* Button size = 40px
* Changed button label

## Testing Instructions
1. Rename a navigation menu
2. Observe the modal matches the 'After' screenshot above.
3. Rename a block without overrides
4. Observe the modal matches the 'After' screenshot above.
5. Rename a block with overrides
6. Observe the modal matches the 'After' screenshot above (the one with the notice).
7. Enable the custom data views experiment and rename a custom view
8. Observe the modal matches the 'After' screenshot above.

## Additional alignment considerations

* Is the 'Name' label worth displaying above the input, or should it be visually hidden? I can see arguments both ways.
* Should the primary button read "Save" or "Rename". In most confirmation dialogs the primary button repeats the action, so I lean towards the latter, but keen to hear thoughts.
